pkg/process/net/windows_pipe.go

```diff
@@ -32,3 +61,17 @@ func (wp *WindowsPipeListener) GetListener() net.Listener {
 func (wp *WindowsPipeListener) Stop() {
 	wp.conn.Close()
 }
+
+// DialSystemProbe connects to the system-probe service endpoint
+func DialSystemProbe(_ string, _ string) (net.Conn, error) {
+	// Unused netType and path
+
+	var timeout = time.Duration(5 * time.Second)
+
+	namedPipe, err := winio.DialPipe(SystemProbePipeName, &timeout)
+	if err != nil {
+		log.Errorf("error connecting to named pipe %s: %s", SystemProbePipeName, err)
+	}
+
+	return namedPipe, err
```

Comment:
since the error is returned, is there a reason to log it? the caller isn't logging it?

Reply:
The existing higher level code still assume socket related errors.
This log makes it clear that the root failure came from a named pipe.

Reply:
> The existing higher level code (a few places) still assumes socket related errors.

what do you mean?

i'll be surprised that the error won't be logged by any of the callers
I suspect that currently we will have multiple logs with the same error

Reply:
The following is common code calling the function and logging the error:
```
	conn, err := net.NewSystemProbeListener(cfg.SocketAddress)
	if err != nil {
		return fmt.Errorf("error creating IPC socket: %s", err)
	}
```

In this snippet, the existing error handler assumes a socket related error, which was fine because a socket was used for all OS platforms.  That is not the case anymore with this change.
That error handler would show something like:
`error creating IPC socket: \\.\pipe\dd_system_probe`,
which is not very sensical.  For someone triaging the log entry without context or tribal knowledge would be confuse.
The error log entry specifically stating that a named pipe connection fail will clarity this ambiguity bettter.

Frankly, I am a bit shocked that a simple error log entry would be scrutinized (versus other existing anti-patterns).

Reply:
Logging is considered "handling" the error, so logging and returning is usually a bad idea because the error would be "handled" twice. The better alternative is to wrap the error. I suggest something like:
```
if err != nil {
    return nil, fmt.Errorf("named pipe %s: %w", SystemProbePipeName, err)
}
```

Reply:
Also, any reason you cannot change the common code error to be `error creating IPC: %s` and moving `socket` or `named pipe` deeper where appropriate?

Reply:
I was aiming to minimize churn as much as possible, but if it is fine, I can move the specialized error messages deeper and make it less ambiguous.  Thanks!

Reply:
you're still logging & returning an error